api/transport/header.go

```diff
@@ -102,6 +111,20 @@ func (h Headers) Items() map[string]string {
 	return h.items
 }
 
+// ForwardingItems returns the non-canonicalized version of the underlying map
+// for this Headers object. The returned map MUST NOT be changed.
+// Doing so will result in undefined behavior.
+func (h Headers) ForwardingItems() map[string]string {
+	if h.keyMapping == nil {
+		return emptyMap
+	}
+	m := make(map[string]string, len(h.keyMapping))
```

Comment:
Actually, if the intention is to call this on the hot-path, you may want to make the internal representation match the result and not copy the map. So, store two copies of the data: `canonicalItems` and `originalItems` instead of `items` and `originalNames`.

And we can probably call this `OriginalItems()` instead.

Reply:
`OriginalItems` is a fine name. In my review I suggested `ExactCaseItems`.

Reply:
going with `OriginalItems` and reverted back to make the internal representation match the result

Reply:
@kriskowal @abhinav If we are going with `ExactCaseItems`, I think we would treat `foo-bar` differently than `Foo-bar` in the `ExactCaseItems` map b/c they are not exact case. This means `ExactCaseItems` might return things like `map[string]string{"A":"b", "a":"c"}`